mlir/lib/Catalyst/Transforms/BufferizationPatterns.cpp

```diff
@@ -37,12 +38,56 @@ struct BufferizePrintOp : public OpConversionPattern<PrintOp> {
     }
 };
 
+struct BufferizeCustomCallOp : public OpConversionPattern<CustomCallOp> {
+    using OpConversionPattern::OpConversionPattern;
+
+    LogicalResult matchAndRewrite(CustomCallOp op, OpAdaptor adaptor,
+                                  ConversionPatternRewriter &rewriter) const override
+    {
+        // Add bufferized arguments
+        SmallVector<Value> bufferArgs;
+        auto operands = adaptor.getOperands();
+        for (auto operand : operands) {
+            bufferArgs.push_back(operand);
+        }
+
+        // Add bufferized return values to the arguments
+        auto results = op.getResults();
```

Comment:
This might be something we should talk about in a style guide, but personally I don't like to obfuscate the type in C++ unless it is already present in the same statement (e.g. a cast) or it is very long. The reason is that when modifying or adding to existing code with obfuscated types you have to wait on the IDE to deduce the type for you, and duck typing is not as reliable/practical in C++ as it is in Python.

But since we don't have an official style guide yet, this is just a suggestion at this point :)

Reply:
Ok no worries, I will remove the auto as much as possible :+1: